static/src/javascripts/projects/common/modules/weather.js

```diff
@@ -0,0 +1,236 @@
+/**
+    "WEATHER"
+
+    Whether the weather be fine,
+    Or whether the weather be not,
+    Whether the weather be cold,
+    Or whether the weather be hot,
+    We'll weather the weather
+    Whatever the weather,
+    Whether we like it or not!
+
+    Author: Anonymous British
+ */
+
+define([
+    'bean',
+    'raven',
+    'common/utils/_',
+    'common/utils/$',
+    'common/utils/ajax',
+    'common/utils/config',
+    'common/utils/mediator',
+    'common/utils/template',
+    'common/modules/search-tool',
+    'common/modules/userPrefs',
+    'text!common/views/weather.html'
+], function (
+    bean,
+    raven,
+    _,
+    $,
+    ajax,
+    config,
+    mediator,
+    template,
+    SearchTool,
+    userPrefs,
+    weatherTemplate
+    ) {
+
+    var self           = null,
+        $weather       = null,
+        $holder        = null,
+        searchTool     = null,
+        prefName       = 'weather-location',
+        getGeoStates   = {
+            process: 'Getting location...',
+            error: 'Unable to get location...',
+            defaultmsg: 'Detect my location'
+        },
+        weatherApiUrl  = '/weather/city',
+        locationApiUrl = '/weather/locations?query=';
```

Comment:
Consider moving these to application config properties: `common/conf/env` rather than hard coding.